src/plugins/index_pattern_editor/public/components/index_pattern_editor_flyout_content.tsx

```diff
@@ -225,52 +238,28 @@ const IndexPatternEditorFlyoutContentComponent = ({
       let newRollupIndexName: string | undefined;
 
       const fetchIndices = async (query: string = '') => {
+        const currentRequestRef = {};
+        currentLoadingMatchedIndicesRef.current = currentRequestRef;
         setIsLoadingMatchedIndices(true);
-        const indexRequests = [];
-
-        if (query?.endsWith('*')) {
-          const exactMatchedQuery = getIndices({
-            http,
-            isRollupIndex,
-            pattern: query,
-            showAllIndices: allowHidden,
-            searchClient,
-          });
-          indexRequests.push(exactMatchedQuery);
-          // provide default value when not making a request for the partialMatchQuery
-          indexRequests.push(Promise.resolve([]));
-        } else {
-          const exactMatchQuery = getIndices({
-            http,
-            isRollupIndex,
-            pattern: query,
-            showAllIndices: allowHidden,
-            searchClient,
-          });
-          const partialMatchQuery = getIndices({
-            http,
-            isRollupIndex,
-            pattern: `${query}*`,
-            showAllIndices: allowHidden,
-            searchClient,
-          });
-
-          indexRequests.push(exactMatchQuery);
-          indexRequests.push(partialMatchQuery);
-        }
-
-        const [exactMatched, partialMatched] = (await ensureMinimumTime(
-          indexRequests
-        )) as MatchedItem[][];
 
-        const matchedIndicesResult = getMatchedIndices(
-          allSources,
-          partialMatched,
-          exactMatched,
-          allowHidden
-        );
-
-        if (isMounted.current) {
+        const { matchedIndicesResult, exactMatched } = !isLoadingSources
+          ? await loadMatchedIndices(query, allowHidden, allSources, {
```

Comment:
Had to extract `loadMatchedIndices` into a separate function to wrap it into `memoize-one`. Otherwise, eslint-hooks complained that it can't resolve args. 
I kept it in the current file to keep it simpler.